### PR TITLE
Upgrades schema-registry to latest and bases on our Java image

### PR DESCRIPTION
**Pull request by adriancole**
As our base java image is 100m, this makes a 77M layer over that
instead of 286M of unshared layers.

```
hypertrace/schema-registry                test                    a0b5e056dcba        10 minutes ago      177MB
hypertrace/schema-registry                0.1.2                   2da89268020e        5 weeks ago         286MB
```

### CODE REVIEW

**Comment by JBAhire** [comment]
Note: @adriancole , we should fix `CYGWIN: not found` error as well.

**Comment by adriancole** [comment]
still have a glitch in the log config, but the cygwin is no longer a problem

**Comment by adriancole** [comment]
@JBAhire @laxmanchekka PTAL logs are down to minimal now

Ex the below is a fine warning in docker-compose as we are using auto-create topic (it is reasonable to not kill this message)
```
docker logs schema-registry
[2020-08-27 13:10:50,588] WARN Creating the schema topic _schemas using a replication factor of 1, which is less than the desired one of 3. If this is a production environment, it's crucial to add more brokers and increase the replication factor of the topic. (io.confluent.kafka.schemaregistry.storage.KafkaStore:203)
```

**Comment by adriancole** [comment]
@laxmanchekka also once this is in and later when changing helm it is important to know that this is using JAVA_OPTS for memory like the other processes do, so this change is needed:

```diff
diff --git a/helm/templates/deployment.yaml b/helm/templates/deployment.yaml
index 6741a27..7deb0bc 100644
--- a/helm/templates/deployment.yaml
+++ b/helm/templates/deployment.yaml
@@ -63,6 +63,8 @@ spec:
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
           env:
+            - name: JAVA_OPTS
+              value: "{{ .Values.heapOptions }}"
             - name: SCHEMA_REGISTRY_HOST_NAME
               valueFrom:
                 fieldRef:
@@ -75,8 +77,6 @@ spec:
               value: {{ template "schema-registry.groupId" . }}
             - name: SCHEMA_REGISTRY_MASTER_ELIGIBILITY
               value: "true"
-            - name: SCHEMA_REGISTRY_HEAP_OPTS
-              value: "{{ .Values.heapOptions }}"
             - name: SCHEMA_REGISTRY_AVRO_COMPATIBILITY_LEVEL
               value: "{{ .Values.avroCompatibilityLevel }}"
             {{- range $configName, $configValue := .Values.configurationOverrides }}
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -98,12 +103,12 @@ workflows:
           filters:
             branches:
               only:
-                - master
+                - main
```
Comment (aaron-steinfeld) [inline .circleci/config.yml:106]: Could have sworn I updated all these...

**Comment by adriancole** [comment]
thx @aaron-steinfeld I don't mind switching this to not define JAVA_OPTS (good point about JAVA_TOOL_OPTIONS, which seems better fit, even if less known).

I might remove it (JAVA_OPTS) anyway, I had a dream of how to clear the errors differently so we don't need to overwrite these shell scripts. basically I am thinking of faking the commands they expect with exported aliases. will give a try and hopefully reduce the changes here (and reuse elsewhere we see cygwin errors etc)

**Comment by adriancole** [comment]
glad I did that as using java's options for things like heap seems quite the rabbit hole https://bugs.openjdk.java.net/browse/JDK-4971166

**Comment by adriancole** [comment]
@aaron-steinfeld Aside, but I don't think we should routinely use JAVA_TOOL_OPTIONS variable as it is made to push down agent config as opposed to routine things https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#tooloptions

**Comment by adriancole** [comment]
forked the JAVA_OPTS thing here as notably I think backfitting to JAVA_TOOL_OPTIONS is going to interfere later if people try to add agent and then be surprised they wiped out other config in doing so. Moreover, it cements a misunderstanding of that variable and as we are agent authors, we should probably be careful about that. https://github.com/hypertrace/hypertrace-gradle-docker-plugins/issues/17

**Comment by adriancole** [comment]
OK it is all sorted now, and requires no change the the helm config later as we are using the native schema-registry scripts again

File: .circleci/config.yml (hunk 1 above)
Comment (adriancole) [reply .circleci/config.yml:106]: probably bc this branch is old

File: docker/install [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+confluent-$CONFLUENT_VERSION/etc/schema-registry \
+confluent-$CONFLUENT_VERSION/share/java/confluent-common \
+confluent-$CONFLUENT_VERSION/share/java/rest-utils \
+confluent-$CONFLUENT_VERSION/share/java/schema-registry
+
+mv confluent-$CONFLUENT_VERSION/* .
+
+# Remove bash as our images doesn't have it, and it isn't required
+sed -i 's~#!/bin/bash~#!/bin/sh~g' bin/*
+# Remove bash syntax relating to irrelevant CYGWIN (prevents console errors)
```
Comment (adriancole) [inline docker/install:18]: @JBAhire @kotharironak @laxmanchekka @ravisingal I think we have some other images that will be able to use this to kill cygwin errors as many people copy/paste the same basic start scripts

**Comment by aaron-steinfeld** [comment]
> forked the JAVA_OPTS thing here as notably I think backfitting to JAVA_TOOL_OPTIONS is going to interfere later if people try to add agent and then be surprised they wiped out other config in doing so. Moreover, it cements a misunderstanding of that variable and as we are agent authors, we should probably be careful about that. [hypertrace/hypertrace-gradle-docker-plugins#17](https://github.com/hypertrace/hypertrace-gradle-docker-plugins/issues/17)

👍 let me see if I can get that in before we do a wider rollout of the plugin updates.

**Pull request merged**